apollo-runtime/src/main/java/com/apollographql/android/impl/RealApolloCall.java

```diff
@@ -10,79 +10,64 @@
 import java.util.Map;
-import java.util.Set;
 import java.util.concurrent.ExecutorService;
 
 import javax.annotation.Nonnull;
 import javax.annotation.Nullable;
 
 import okhttp3.Call;
 import okhttp3.HttpUrl;
-import okhttp3.internal.Util;
 
 import static com.apollographql.android.api.graphql.util.Utils.checkNotNull;
 
-@SuppressWarnings("WeakerAccess") final class RealApolloCall<T> extends BaseApolloCall implements ApolloCall<T> {
-  volatile Call httpCall;
-  private final Cache cache;
-  CacheControl cacheControl = CacheControl.CACHE_FIRST;
-  private final HttpCache httpCache;
-  HttpCacheControl httpCacheControl = HttpCacheControl.CACHE_FIRST;
-  private final ResponseFieldMapper responseFieldMapper;
-  private final Map<ScalarType, CustomTypeAdapter> customTypeAdapters;
-  private final ExecutorService dispatcher;
-  private boolean executed;
-
-  RealApolloCall(Operation operation, HttpUrl serverUrl, Call.Factory httpCallFactory, HttpCache httpCache, Moshi moshi,
-      ResponseFieldMapper responseFieldMapper, Map<ScalarType, CustomTypeAdapter> customTypeAdapters, Cache cache,
-      ExecutorService dispatcher) {
-    super(operation, serverUrl, httpCallFactory, moshi);
-    this.httpCache = httpCache;
-    this.responseFieldMapper = responseFieldMapper;
-    this.customTypeAdapters = customTypeAdapters;
-    this.cache = cache;
-    this.dispatcher = dispatcher;
-  }
-
-  private RealApolloCall(Operation operation, HttpUrl serverUrl, Call.Factory httpCallFactory, HttpCache httpCache,
+@SuppressWarnings("WeakerAccess") final class RealApolloCall<T> implements ApolloCall<T> {
+  final Operation operation;
+  final HttpUrl serverUrl;
+  final Call.Factory httpCallFactory;
+  final HttpCache httpCache;
+  final HttpCacheControl httpCacheControl;
+  final Moshi moshi;
+  final ResponseFieldMapper responseFieldMapper;
+  final Map<ScalarType, CustomTypeAdapter> customTypeAdapters;
+  final Cache cache;
+  final CacheControl cacheControl;
+  final CallInterceptorChain interceptorChain;
+  final ExecutorService dispatcher;
+  volatile boolean executed;
+
+  RealApolloCall(Operation operation, HttpUrl serverUrl, Call.Factory httpCallFactory, HttpCache httpCache,
       HttpCacheControl httpCacheControl, Moshi moshi, ResponseFieldMapper responseFieldMapper,
       Map<ScalarType, CustomTypeAdapter> customTypeAdapters, Cache cache, CacheControl cacheControl,
       ExecutorService dispatcher) {
-    super(operation, serverUrl, httpCallFactory, moshi);
+    this.operation = operation;
+    this.serverUrl = serverUrl;
+    this.httpCallFactory = httpCallFactory;
     this.httpCache = httpCache;
     this.httpCacheControl = httpCacheControl;
+    this.moshi = moshi;
     this.responseFieldMapper = responseFieldMapper;
     this.customTypeAdapters = customTypeAdapters;
     this.cache = cache;
     this.cacheControl = cacheControl;
     this.dispatcher = dispatcher;
+
+    interceptorChain = new RealCallInterceptorChain(operation)
```

Comment:
How about this to change API from `com.apollographql.android.impl.RealCallInterceptorChain#chain` to `addIntercetor`